maliang/standard/widgets.py

```diff
@@ -1458,7 +1458,7 @@ def __init__(
             underline=underline, overstrike=overstrike, anchor=anchor, limit_width=-h)
         self._button = Button(
             self, (size[0]-h-5-self.offset[0], 5-self.offset[1]), (h, h), text="▼",
-            command=self._open_options)
+            command=lambda: self._open_options() if self._segmented_button.disappeared else self._close_options())
```

Comment:
```suggestion
            command=lambda: self._open_options() if self._segmented_button.disappeared else self._segmented_button.forget())
```

此处不能直接调用方法 `self._close_options`，因为方法 `self._close_options` 会在关闭选择框的同时修改当前的值。如果你已经选择了一个选项，你直接关闭会导致当前的值被清空，而这是不对的。